File: framework/security/src/main/java/top/bettercode/simpleframework/security/UserDetailsAuthenticationProvider.java
package top.bettercode.simpleframework.security;

import org.springframework.security.authentication.AuthenticationProvider;
import org.springframework.security.core.Authentication;
import org.springframework.security.core.AuthenticationException;

public final class UserDetailsAuthenticationProvider implements AuthenticationProvider {


  @Override
  public Authentication authenticate(Authentication authentication) throws AuthenticationException {
    authentication.setAuthenticated(true);
    return authentication;
  }


  @Override
  public boolean supports(Class<?> authentication) {
    return UserDetailsAuthenticationToken.class.isAssignableFrom(authentication);
  }
}

File: security/src/main/kotlin/top/bettercode/summer/security/authorization/UserDetailsAuthenticationProvider.kt
package top.bettercode.summer.security.authorization

import org.springframework.security.authentication.AuthenticationProvider
import org.springframework.security.core.Authentication

class UserDetailsAuthenticationProvider : AuthenticationProvider {
    override fun authenticate(authentication: Authentication): Authentication {
        authentication.isAuthenticated = true
        return authentication
    }

    override fun supports(authentication: Class<*>): Boolean {
        return UserDetailsAuthenticationToken::class.java.isAssignableFrom(authentication)
    }
}
